### DISCUSSION TITLE
Is there a definitive tutorial for using this Repo? And a few specific questions regarding training.

### DISCUSSION THREAD

**Discussion by PeterLorre** [TheLastBen/fast-stable-diffusion#1304] [General]
Between the discussions here and on Reddit, there are a lot of great pointers and tips. However, has anyone actually made a definitive and robust tutorial for this training colab? I still have so many questions, and I apologize if they have been previously answered.

1. What does text encoding actually do? 
2. Do the image captions significantly help the training of the subject? If so, how? For instance, A photo of a middle age woman drinking a cup of tea in a busy restaurant. Does this description actually help train the model along with the instance images?
3. If I'm training a person, and I want their face as well as body shape / clothing, should I be using full body images as long as they are resized and cropped to the corresponding sizes? e.g. 512-768
4. If I have 20 instance images of a person, and I want their face and likeness, could I also upload hundreds of concept images that would train along side of that? Or would I first train the person to the correct amount, and then train concept images after that and set the Unet and Text encoder to zero?
5. Finally, regarding the style training option, it says "check it after training for at least 1000 steps". Is that referring to 1000 concept training steps or UNet training steps? 

Thank you for all of your legendary work in making this freely available! Much appreciated.

**Comment by TheLastBen** [comment]
- The text encoder translates (encodes) the prompt into a language the model understands, finetuning it on specific images will make the model converge faster to their likeness.
- Captions help only when finetuning on a large scale to get a better model, you don't need them when you train on only a few instances.
- When you train on a person, it is preferable to have a mix of closeup and longshots but more closeups to get the face correctly, and also try using images with less complicated background, a simple color is preferable.
- When you train on a person, you don't need captions and you don't need concept images, they are suited for large finetuning and style training.
- The style training option is for the UNet, it trains on the instance images, but without putting too much weight on the instance name (identifier) to avoid excess overfitting.
